test/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -188,7 +165,9 @@ end
             n = 10
 
             @testset "eigen!(::Matrix{$T}) frule" for T in (Float64,ComplexF64)
-                X = randn(T, n, n)
+
+                # get a bit away from zero so don't have finite differencing woes
+                X = randn(T, n, n) .+ 5.0
```

Comment:
I was getting these huge values like 10^5 in the output of FDM.
Fixing the seed fixes it, but this seems to fix it robustly.
I am guessing there are some values of X that fdm doesn't like,
and before we were lucky enough to not hit them.
but since we are using the RNG a different number of times now, we are not anymore.

Similar happened below

Reply:
Hmm interesting. @wesselb or @sethaxen might know what's going on? My suspicion would be that the eigenvalues are flipping around or something? If so, might help to find a way to space them out more or something.

Reply:
> My suspicion would be that the eigenvalues are flipping around or something?

It could be, but the major source of instability I saw was in the phase of the eigenvectors. The convention LinearAlgebra uses (from LAPACK) is that the entry with the largest magnitude in each eigenvector is rotated to be positive real or negative real, whichever is nearest. So it could become unstable if another entry was close to that one in magnitude. So one way might be to do something like this:
```julia
V = hcat(Random.shuffle([normalize((x .+ 1e-2 .* randn.(ComplexF64)) * sign(randn(ComplexF64))) for x in eachcol(Matrix{Float64}(I, n, n))])...)
```
which creates an identity matrix, adds small random complex numbers to all entries, applies a random phase convention to each eigenvector, then normalizes them. The eigenvectors are then shuffled and made columns of a new matrix. This produces one where the element with the largest magnitude is well-separated from the rest. With a suitable set of eigenvalues, perhaps this would make a well-behaved primal.

Reply:
Shall we open issue about this,  and merge this PR as is?

Reply:
I would be in favour of that. Let's get this in.

Reply:
hmm still failing on some OS for some julia versions.
Which is annoying, something is funky happening with the RNG

```suggestion
                X = 10 .* (randn(T, n, n) .+ 5.0)
```